plugins/auth_legacy/src/native.cpp

```diff
@@ -1,57 +1,59 @@
-// =-=-=-=-=-=-=-
-// irods includes
-#include "irods/rodsDef.h"
-#include "irods/msParam.h"
-#include "irods/rcConnect.h"
-#include "irods/authRequest.h"
-#include "irods/authResponse.h"
 #include "irods/authCheck.h"
-#include "irods/miscServerFunct.hpp"
 #include "irods/authPluginRequest.h"
+#include "irods/authRequest.h"
+#include "irods/authResponse.h"
 #include "irods/authenticate.h"
-#include "irods/rsAuthCheck.hpp"
-#include "irods/rsAuthRequest.hpp"
-
-// =-=-=-=-=-=-=-
-#include "irods/irods_auth_plugin.hpp"
 #include "irods/irods_auth_constants.hpp"
+#include "irods/irods_auth_plugin.hpp"
+#include "irods/irods_kvp_string_parser.hpp"
 #include "irods/irods_native_auth_object.hpp"
 #include "irods/irods_stacktrace.hpp"
-#include "irods/irods_kvp_string_parser.hpp"
+#include "irods/miscServerFunct.hpp"
+#include "irods/msParam.h"
+#include "irods/rcConnect.h"
+#include "irods/rodsDef.h"
+#include "irods/rsAuthCheck.hpp"
+#include "irods/rsAuthRequest.hpp"
 
-// =-=-=-=-=-=-=-
-// stl includes
+#include <iostream>
 #include <sstream>
 #include <string>
-#include <iostream>
+#include <string_view>
 #include <termios.h>
 #include <unistd.h>
 
 #include <openssl/md5.h>
 
+#include "fmt/format.h"
+
 int get64RandomBytes( char *buf );
 void setSessionSignatureClientside( char* _sig );
 void _rsSetAuthRequestGetChallenge( const char* _c );
 
-static irods::error check_proxy_user_privileges(
-    rsComm_t *rsComm,
-    int proxyUserPriv ) {
-    irods::error result = SUCCESS();
-
-    if ( strcmp( rsComm->proxyUser.userName, rsComm->clientUser.userName ) != 0 ) {
-
-        /* remote privileged user can only do things on behalf of users from
-         * the same zone */
-        result = ASSERT_ERROR( proxyUserPriv >= LOCAL_PRIV_USER_AUTH ||
-                               ( proxyUserPriv >= REMOTE_PRIV_USER_AUTH &&
-                                 strcmp( rsComm->proxyUser.rodsZone, rsComm->clientUser.rodsZone ) == 0 ),
-                               SYS_PROXYUSER_NO_PRIV,
-                               "Proxyuser: \"%s\" with %d no priv to auth clientUser: \"%s\".",
-                               rsComm->proxyUser.userName, proxyUserPriv, rsComm->clientUser.userName );
-    }
+namespace
+{
+    auto check_proxy_user_privileges(rsComm_t *rsComm, const int proxyUserPriv) -> irods::error
```

Comment:
Might consider moving the version of this function found in the new plugin to a common place and use it in both plugins.

Reply:
one possibility is `irods_rs_comm_query.hpp`.